ledger/accountdb.go

```diff
@@ -262,6 +272,15 @@ func (prd *persistedResourcesData) AccountResource() ledgercore.AccountResource
 	return ret
 }
 
+//msgp:ignore persistedKVData
+type persistedKVData struct {
+	// kv value
+	value *string
```

Comment:
I originally used `string` because I like the immutability.  I caved under the weight of some of our tooling, which expects to serialize bytes as b64, but string as unicode (in json), so I use `[]byte` in some places.

I am worried the switch may be introducing some copies.

Reply:
As I look into changing this, please speak up if you will be horrified by code that distinguishes `nil` from `[]byte{}`. If I switch over to `[]byte` I will probably still use `nil` to mean "this key has been deleted".

Reply:
Resolving - https://github.com/algorand/go-algorand/pull/4686 converts KV values from string to []byte.